frontend/src/metabase/nav/components/WhatsNewNotification/WhatsNewNotification.tsx

```diff
@@ -0,0 +1,92 @@
+import { t } from "ttag";
+import { useCallback, useMemo } from "react";
+import styled from "@emotion/styled";
+import { updateSetting } from "metabase/admin/settings/settings";
+import IconButtonWrapper from "metabase/components/IconButtonWrapper";
+import { Icon } from "metabase/core/components/Icon";
+import { isNotFalsy } from "metabase/core/utils/types";
+import { color } from "metabase/lib/colors";
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import { getIsEmbedded } from "metabase/selectors/embed";
+import { getSetting } from "metabase/selectors/settings";
+import { Flex, Stack, Text, Anchor, Box } from "metabase/ui";
+import { getLatestEligibleReleaseNotes } from "./utils";
+import Sparkles from "./sparkles.svg?component";
+
+export function WhatsNewNotification() {
+  const dispatch = useDispatch();
+  const isEmbedded = useSelector(getIsEmbedded);
+  const versionInfo = useSelector(state => getSetting(state, "version-info"));
+  const currentVersion = useSelector(state => getSetting(state, "version"));
+  const lastAcknowledgedVersion = useSelector(state =>
+    getSetting(state, "last-acknowledged-version"),
+  );
+
+  const url: string | undefined = useMemo(() => {
+    const versionsList = [versionInfo?.latest]
+      .concat(versionInfo?.older)
+      .filter(isNotFalsy);
```

Comment:
this is to make sure we're not sending undefined values while loading (and to make ts happy)

Reply:
Could you please move the setting value and this check to the utility function?